README.md

```diff
@@ -16,45 +16,111 @@ No license is granted to the Wire trademark and its associated logos, all of whi
 by Wire Swiss GmbH. Any use of the Wire trademark and/or its associated logos is expressly prohibited without the
 express prior written consent of Wire Swiss GmbH.
 
-## General testing
+## Parts
+
+* acme: types that deal with ACME certificate enrollment
+* e2e-identity: implementation of the Wire end-to-end identity workflow, built on top of acme and jwt
+* ffi: Haskell bindings for rusty-jwt-tools, only used by wire-server
+* jwt: a collection of JWT utilities
+* x509-check: helpers for X509 certificate validation, only used by acme
+
+## Building
+
+For the build requirements, look at [the information in core-crypto repo](
+https://github.com/wireapp/core-crypto?tab=readme-ov-file#general-requirements).
+
+> [!note]
+> Building rusty-jwt-tools independently of core-crypto for Android targets is currently not supported due to
+> missing configuration bits. However, the necessary bits are in core-crypto so one can build rusty-jwt-tools
+> for Android targets as part of a core-crypto build.
+
+Building is as simple as
+```bash
+cargo build
+```
+
+## Testing
 
 Install cargo-nextest if you haven't done so, it yields some substantial speedup.
-Also, it allows us to group tests easily and have custom settings for those groups (
-see [`.config/nextest.toml`](.config/nextest.toml)).
+Also, it allows us to group tests easily and have custom settings for those groups (see [`.config/nextest.toml`](.config/nextest.toml)).
 
 ```bash
 cargo install cargo-nextest
 ```
 
-Start docker.
-The integration tests are very flaky, mainly because the behavior of the Keycloak docker container is very flaky.
-This is why we need the `no-fail-fast` flag in the test command.
-In the average case, about 75 % of the tests pass.
-A possible approach to deal with the flakiness is to run the command below and then run the tests that failed in the
-first run individually.
+Make sure the docker daemon is running (this is needed because the test suite runs an OIDC provider
+inside a container).
+
+> [!warning]
+> Integration tests are very flaky, mainly because the behavior of the Keycloak docker container is very flaky.
+> This is why we need the `--no-fail-fast` flag in the test command.
+> In the average case, about 75 % of the tests pass.
+> A possible approach to deal with the flakiness is to run the command below and then run the tests that failed
+> in the first run individually.
 
+Run tests:
 ```bash
 cargo nextest run --no-fail-fast
```

Comment:
Could be that I forgot changing this after some keycloak update. IIRC, when run sequentially, the tests reliably pass. Do you know if that's true? If so, we can just mention here that we have to run sequentially, but remove the part about `--no-fail-fast`.